pkg/services/notary/notary.go

```diff
@@ -187,6 +188,24 @@ func (n *Notary) mainLoop() {
 			n.PostPersist()
 		}
 	}
+drainBlocksLoop:
+	for {
+		select {
+		case <-n.blocksCh:
+		default:
+			break drainBlocksLoop
+		}
+	}
+	close(n.blocksCh)
+drainTxsLoop:
```

Comment:
We probably can combine it with the previous one.

Reply:
Yes, fixed.